lib/twilio/rtc/peerconnection.ts

```diff
@@ -95,6 +96,11 @@ function PeerConnection(audioHelper, pstream, options) {
   this.util = options.util || util;
   this.codecPreferences = options.codecPreferences;
 
+  this._audioProcessorEventObserver.on('add', (isRemote) =>
+    isRemote && this._onRemoteAudioProcessorAdded());
+  this._audioProcessorEventObserver.on('remove', (isRemote) =>
+    isRemote && this._onRemoteAudioProcessorRemoved());
```

Comment:
You also need to remove these handlers inside the `close` method, right after calling destroyRemoteProcessedStream. To track the function reference, set the handler directly. Renamed the handler to remove 'Remote' since we'll be doing the checks in the handlers.

```js
this._audioProcessorEventObserver.on('add', this._onAudioProcessorAdded);
this._audioProcessorEventObserver.on('remove', this._onAudioProcessorRemoved);
```

Then use arrow function when defining these
```js

PeerConnection.prototype._onAudioProcessorAdded = (isRemote) => {
  if (!isRemote)
    return;
  }
  // ... fill in the rest
```